runtime/alloc.c

```diff
@@ -371,3 +371,27 @@ CAMLexport value caml_alloc_some(value v)
   Field(some, 0) = v;
   CAMLreturn(some);
 }
+
+
+/* For aligned atomic */
+
+CAML_STATIC_ASSERT((sizeof(value)) <= 8);
+
+CAMLprim value caml_make_atomic_aligned(value val)
+{
+  value var = caml_alloc_shr(7, 0);
+  CAMLassert((var - 8) % Allocation_alignment == 0);
+  caml_initialize(&Field(var, 0), val);
+  for (int i = 1; i < 7; i++)
+    caml_initialize(&Field(var, i), 0);
+  return var;
+}
+
+CAMLprim value caml_atomic_is_aligned(value val)
+{
+  if ((val - 8) % Allocation_alignment == 0) {
```

Comment:
The `8` here needs to be fixed in the same way as `caml_make_atomic_aligned`. `Hp_val` macro may make it explicit what you are trying to do here:

https://github.com/ocaml/ocaml/blob/f11f82f18993f969be5cf5bc4a0c133a053949cb/runtime/caml/mlvalues.h#L162

It returns the pointer to the header given the value on the heap.

Reply:
Change above code code to use `Hp_val` (the function is now defined in the test).